verifier/server/app.js

```diff
@@ -1,6 +1,30 @@
 const express = require('express');
+const { join } = require('path');
 const { auth, resolver, loaders } = require('@iden3/js-iden3-auth');
 const getRawBody = require('raw-body');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+const argv = yargs(hideBin(process.argv)).option('state-contract', { string: true }).argv;
+
+let publicHost = argv['public-host'];
+if (!publicHost) {
+  console.log("No publicly accessible hostname provided with '--public-host', using localhost:8080");
+  publicHost = 'http://localhost:8080';
+} else {
+  publicHost = publicHost.replace(/\/+$/, '');
+}
+
+let jsonrpcUrl = argv['jsonrpc-url'];
+if (!jsonrpcUrl) {
+  console.error('Must provide the URL of the JSON-RPC endpoint for the blockchain that hosts the State contract, with the --jsonrpc-url parameter');
+  process.exit(1);
+}
+
+let stateContract = argv['state-contract'];
+if (!stateContract) {
```

Comment:
@jimthematrix is the contract actually invoked here to get the state to verify against, or are the JSON-RPC URL and State contract address included in the challenge QR code? They don't seem to actually be used, e.g. I tried giving localhost:8081 and the contract address from README, and it generated the same QR code as with proper values.

Or is this just setting the stage for next steps?

Reply:
@nedgar the blockchain URL and the contract address are used during the processing of the callback request. after the holder decoded the QR it prepares the proof based on the requested scopes inside the QR, then the holder calls the `POST /callback` endpoint. The server then verifies the proof inside the response, which requires looking up the latest published state of the issuer. That's when the url and the address are consumed (inside the `EthStateResolver`)

Reply:
Thanks for clarifying, @jimthematrix, that makes sense.